source/extensions/filters/network/mongo_proxy/proxy.cc

```diff
@@ -155,13 +155,14 @@ void ProxyFilter::decodeQuery(QueryMessagePtr&& message) {
     Stats::StatNameVec names;
     names.reserve(6); // 2 entries are added by chargeQueryStats().
     names.push_back(mongo_stats_->collection_);
-    names.push_back(mongo_stats_->getDynamic(active_query->query_info_.collection()));
+    Stats::StatNameDynamicPool dynamic(mongo_stats_->symbolTable());
```

Comment:
@jplevyak PTAL at this change even if you won't be able to review the entire PR. Note taht StatNameSet::getDynamic() is being removed in this PR in preference to this new model of making StatNames from dynamically changing strings that inline all the bytes and do not share or take locks (inspired by our discussion last month).

I imagine some of these sorts of changes will need to happen in the wasm fork.